### PR TITLE
Upgrade Elasticsearch client to 7.x

### PR DESCRIPTION
**Pull request by wendigo**
Release notes:

`Add support for Elasticsearch 8.x and drop support for Elasticsearch 6.x`

### CODE REVIEW

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/ElasticsearchQueryRunner.java [hunk 1]
```diff
@@ -112,12 +134,27 @@ public static DistributedQueryRunner createElasticsearchQueryRunner(
...
     }
 
+    public static SSLContext getSSLContext()
+    {
+        try {
+            return createSSLContext(
+                    Optional.empty(),
+                    Optional.empty(),
+                    Optional.of(new File(getResource("truststore.jks").toURI())),
+                    Optional.of("123456"));
```
Comment (mkhludnev) [inline plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/ElasticsearchQueryRunner.java:144]: `PASSWORD` ?

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/ElasticsearchQueryRunner.java [hunk 2]
```diff
@@ -130,6 +167,13 @@ private static void installElasticsearchPlugin(
                 .put("elasticsearch.scroll-size", "1000")
                 .put("elasticsearch.scroll-timeout", "1m")
                 .put("elasticsearch.request-timeout", "2m")
+                .put("elasticsearch.tls.enabled", "true")
+                .put("elasticsearch.tls.truststore-path", new File(getResource("truststore.jks").toURI()).getPath())
+                .put("elasticsearch.tls.truststore-password", "123456")
```
Comment (mkhludnev) [inline plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/ElasticsearchQueryRunner.java:172]: `PASSWORD`?

**Review by mkhludnev — APPROVED** [review]

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/ElasticsearchQueryRunner.java (hunk 2 above)
Comment (wendigo) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/ElasticsearchQueryRunner.java:172]: We usually use this one for statically generated JKS

**Comment by mkhludnev** [comment]
It's rather cool, @wendigo. So, far it fails on Opensearch and Ngnix.

**Comment by wendigo** [comment]
@mkhludnev yeah, i'm aware. Misconfiguration

**Comment by mkhludnev** [comment]
Hold on, before get into this I encounter #11377, now it seems like it's mandatory. Will Opensearch connector be a separate plugin or we can hope both clients to reside in the same module?

**Comment by martint** [comment]
See https://github.com/trinodb/trino/pull/11295#issuecomment-1058374147

**Comment by wendigo** [comment]
OpenSearch support is moved to a separate connector via https://github.com/trinodb/trino/pull/20257

**Comment by wendigo** [comment]
OpenSearch support is moved to a separate connector via https://github.com/trinodb/trino/pull/20257

**Pull request closed without merging**